### PR TITLE
vue-language-server: init at 2.0.21

### PR DESCRIPTION
**Pull request by friedow**
## Description of changes

This adds the official vue-language-server (called @vue/language-server in npm) package. This is needed for example for the neovim lsp integration (https://github.com/vuejs/language-tools?tab=readme-ov-file#community-integration).

Furethermore this PR removes the deprecated nodePackages.@volar/vue-language-server.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by a-kenji** [comment]
Result of `nixpkgs-review pr 319134` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>lv_img_conv</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
    <li>postcss-cli</li>
    <li>wrangler</li>
  </ul>
</details>
<details>
  <summary>42 packages built:</summary>
  <ul>
    <li>akkoma-frontends.akkoma-fe</li>
    <li>balena-cli</li>
    <li>bat-extras.prettybat</li>
    <li>code-server</li>
    <li>commitlint</li>
    <li>concurrently</li>
    <li>create-cycle-app</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>documenso</li>
    <li>emanote</li>
    <li>emanote.data</li>
    <li>emanote.doc</li>
    <li>emojione</li>
    <li>fast-cli</li>
    <li>image_optim</li>
    <li>joplin</li>
    <li>lemmy-ui</li>
    <li>lessc</li>
    <li>lighthouse</li>
    <li>nodehun</li>
    <li>openvscode-server</li>
    <li>pscid</li>
    <li>styx</li>
    <li>styx.lib</li>
    <li>styx.themes</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>uppy-companion</li>
    <li>vimPlugins.coc-docker</li>
    <li>vimPlugins.coc-explorer</li>
    <li>vimPlugins.coc-go</li>
    <li>vimPlugins.coc-metals</li>
    <li>vimPlugins.coc-pyright</li>
    <li>vimPlugins.coc-sh</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.coc-sumneko-lua</li>
    <li>vimPlugins.coc-tslint</li>
    <li>vimPlugins.coc-vetur</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
    <li>wasm-text-gen</li>
  </ul>
</details>

**Comment by lelgenio** [comment]
Could this be done using [buildNpmPackage](https://nixos.org/manual/nixpkgs/stable/#javascript-buildNpmPackage)?

**Comment by a-kenji** [comment]
Result of `nixpkgs-review pr 319134` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>zerobin</li>
    <li>zerobin.dist</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>lv_img_conv</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
    <li>postcss-cli</li>
    <li>wrangler</li>
  </ul>
</details>
<details>
  <summary>44 packages built:</summary>
  <ul>
    <li>akkoma-frontends.akkoma-fe</li>
    <li>balena-cli</li>
    <li>bat-extras.prettybat</li>
    <li>code-server</li>
    <li>commitlint</li>
    <li>concurrently</li>
    <li>create-cycle-app</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>documenso</li>
    <li>emanote (haskellPackages.emanote)</li>
    <li>emanote.data (haskellPackages.emanote.data)</li>
    <li>emanote.doc (haskellPackages.emanote.doc)</li>
    <li>emojione</li>
    <li>fast-cli</li>
    <li>haskellPackages.tailwind</li>
    <li>haskellPackages.tailwind.doc</li>
    <li>image_optim</li>
    <li>joplin</li>
    <li>lemmy-ui</li>
    <li>lessc</li>
    <li>lighthouse</li>
    <li>nodehun</li>
    <li>openvscode-server</li>
    <li>pscid</li>
    <li>styx</li>
    <li>styx.lib</li>
    <li>styx.themes</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>uppy-companion</li>
    <li>vimPlugins.coc-docker</li>
    <li>vimPlugins.coc-explorer</li>
    <li>vimPlugins.coc-go</li>
    <li>vimPlugins.coc-metals</li>
    <li>vimPlugins.coc-pyright</li>
    <li>vimPlugins.coc-sh</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.coc-sumneko-lua</li>
    <li>vimPlugins.coc-tslint</li>
    <li>vimPlugins.coc-vetur</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
    <li>wasm-text-gen</li>
  </ul>
</details>

**Review by a-kenji — APPROVED** [review]
LGTM, tested basic ipc of the lsp.

**Comment by a-kenji** [comment]
That could be a good way do build it @lelgenio.
It seems it is in the npm registry: https://www.npmjs.com/package/@vue/language-server

**Comment by a-kenji** [comment]
@friedow 
Please change the commit message and title of the pr to:

nodePackages.@vue/language-server: init at [VERSION]

**Comment by friedow** [comment]
1. Should I move this into a 'standalone package' build with [buildNpmPackage](https://nixos.org/manual/nixpkgs/stable/#javascript-buildNpmPackage) or not? This would differ from the other nodePackages, is that fine?
2. I think the packages that are failing to build fail because the [generate.sh](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/generate.sh) script for nodePackages updates the package.lock file. This effectively updates all nodePackages, resulting in broken packages that depend on those. How do we deal with something like that in nixpkgs?
3. The question from my note above is still open: This package should replace @volar/vue-language-server. How do we deal with deprecations in nixpkgs? Should I remove @volar/vue-language-server in another PR?

**Comment by SuperSandro2000** [comment]
> This would differ from the other nodePackages, is that fine?

We don't want any new nodePackages as they are a pain to update and maintain.

> This effectively updates all nodePackages, resulting in broken packages that depend on those. How do we deal with something like that in nixpkgs?

We don't, nodePackages is broken by design and if you care enough about a package in it, that it being broken is a problem, it should be packaged separately.

> This package should replace @volar/vue-language-server. How do we deal with deprecations in nixpkgs? Should I remove @volar/vue-language-server in another PR?

Yes, same PR. See the readme in nodePackages and aliases file for node.

**Review by SuperSandro2000 — DISMISSED** [review]
See above comment

**Comment by friedow** [comment]
@SuperSandro2000 thanks for clarifying :), I'll package this as a separate package using buildNpmPackage then. Furthermore, in the same PR I'll remove @volar/vue-language-server. Will do that later today ;).

**Comment by friedow** [comment]
@SuperSandro2000, @a-kenji , @lelgenio I've updated this PR according to your suggestions. This PR now includes vue-language-server (based on [@vue/language-server in npm](https://www.npmjs.com/package/@vue/language-server)) as a standalone package build with `buildNpmPackage`. Furthermore, it removes the deprecated [nodepackages.@volar/vue-language-server](https://www.npmjs.com/package/@volar/vue-language-server). 
Would you take another look?

**Comment by lelgenio** [comment]
You could add an alias in `pkgs/development/node-packages/aliases.nix`, so it still works for people using `nodePackages."@volar/vue-language-server"`.

**Comment by lelgenio** [comment]
Result of `nixpkgs-review pr 319134` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>zerobin</li>
    <li>zerobin.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>lv_img_conv</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
    <li>postcss-cli</li>
    <li>vscode-js-debug</li>
    <li>wrangler</li>
  </ul>
</details>
<details>
  <summary>48 packages built:</summary>
  <ul>
    <li>akkoma-frontends.akkoma-fe</li>
    <li>balena-cli</li>
    <li>bat-extras.prettybat</li>
    <li>code-server</li>
    <li>commitlint</li>
    <li>concurrently</li>
    <li>create-cycle-app</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>documenso</li>
    <li>emanote (haskellPackages.emanote)</li>
    <li>emanote.data (haskellPackages.emanote.data)</li>
    <li>emanote.doc (haskellPackages.emanote.doc)</li>
    <li>emojione</li>
    <li>fast-cli</li>
    <li>haskellPackages.tailwind</li>
    <li>haskellPackages.tailwind.doc</li>
    <li>image_optim</li>
    <li>imapnotify</li>
    <li>joplin</li>
    <li>lemmy-ui</li>
    <li>lessc</li>
    <li>lighthouse</li>
    <li>nodehun</li>
    <li>openvscode-server</li>
    <li>pscid</li>
    <li>styx</li>
    <li>styx.lib</li>
    <li>styx.themes</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.plugins.giphy</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>uppy-companion</li>
    <li>vimPlugins.coc-clangd</li>
    <li>vimPlugins.coc-docker</li>
    <li>vimPlugins.coc-explorer</li>
    <li>vimPlugins.coc-go</li>
    <li>vimPlugins.coc-metals</li>
    <li>vimPlugins.coc-pyright</li>
    <li>vimPlugins.coc-sh</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.coc-sumneko-lua</li>
    <li>vimPlugins.coc-tslint</li>
    <li>vimPlugins.coc-vetur</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
    <li>vue-language-server</li>
    <li>wasm-text-gen</li>
  </ul>
</details>

File: pkgs/by-name/vu/vue-language-server/package.nix
Comment (lelgenio) [inline pkgs/by-name/vu/vue-language-server/package.nix:1]: Consider formatting using `nixfmt-rfc-style`

Comment (friedow) [reply pkgs/by-name/vu/vue-language-server/package.nix:1]: done

**Comment by friedow** [comment]
@lelgenio both done: aliases are added, nixfmt-rfc-style was executed

**Review by a-kenji — APPROVED** [review]

**Review by lelgenio — APPROVED** [review]

**Review by lelgenio — CHANGES_REQUESTED** [review]
`pkgs/development/node-packages/overrides.nix` still has a reference to the old volar package

**Review by lelgenio — APPROVED** [review]

**Comment by friedow** [comment]
@SuperSandro2000 could you review again?

**Comment by friedow** [comment]
@SuperSandro2000, I dont wanna annoy you but could you review again please?

File: pkgs/development/node-packages/node-env.nix [hunk 1]
```diff
@@ -496,7 +496,7 @@ let
     let
       extraArgs = removeAttrs args [ "name" "dependencies" "buildInputs" "dontStrip" "dontNpmInstall" "preRebuild" "unpackPhase" "buildPhase" "meta" ];
     in
-    stdenv.mkDerivation (finalAttrs: {
```
Comment (Artturin) [inline pkgs/development/node-packages/node-env.nix:499]: @Scrumplex this reverted your https://github.com/NixOS/nixpkgs/commit/4549773edf119c40f3285fd46b662a5ceec2e456

**Pull request merged**